core/execution/engine.go

```diff
@@ -794,6 +799,19 @@ func (e *Engine) OnMarketAuctionMinimumDurationUpdate(ctx context.Context, d tim
 	return nil
 }
 
+func (e *Engine) OnMarkPriceUpdateMaximumFrequency(ctx context.Context, d time.Duration) error {
+	// we make sure to update both the copy and the actual markets for snapshots
+	// although we can most likely just update the market because we're already ensuring the nextMTM value
+	// is set correctly when getting state
+	for _, cpMkt := range e.marketsCpy {
+		mkt := e.markets[cpMkt.mkt.ID]
+		mkt.OnMarkPriceUpdateMaximumFrequency(ctx, d)
+		cpMkt.nextMTM = mkt.nextMTM
```

Comment:
I don't think we need to worry about making sure `e.marketsCpy` is also updated, because its just a pointer to the same market?

Reply:
Unfortunately, not pointers to the same object.

Reply:
Are you sure? in the engine they're defined like his:
```
markets    map[string]*Market
marketsCpy []*Market
```

and when we add a market we do this:
```
e.markets[marketConfig.ID] = mkt
e.marketsCpy = append(e.marketsCpy, mkt)
```
?

Reply:
marketsCpy definitely points to the same objects, they just exists to iterate deterministically over the markets without sorting them everytime.

Reply:
alright, don't know why I thought otherwise, I'll fix that up && push